float32 amp_in
float32 neu_in
float32 phi_in
float32 output
